server/runtime/src/main/java/org/infinispan/server/Extensions.java

```diff
@@ -86,4 +93,16 @@ private void loadServerTasks(ClassLoader classLoader) {
    public TaskEngine getServerTaskEngine(EmbeddedCacheManager cm) {
       return new ServerTaskEngine(cm, serverTasks);
    }
+
+   public void loadSchemas(String path, ProtobufMetadataManager protobufMetadataManager) throws Exception {
+      Path schemaPath = Paths.get(path);
+      if (schemaPath.toFile().exists()) {
+         List<Path> schemas = Files.walk(schemaPath)
+               .filter(f -> f.toString().endsWith(".proto"))
+               .collect(Collectors.toList());
+         for (Path schema : schemas) {
+            protobufMetadataManager.registerProtofile(schema.toString(), new String(Files.readAllBytes(schema)));
```

Comment:
If there are any errors (after processing all the files) I would dump to log the names of the offending files and their errors and prevent the server from starting

Reply:
Unless the `schema` directory is mounted on a persistent volume, similar to how the `data` directory is